docs/source/tutorials/chat.rst

```diff
@@ -0,0 +1,164 @@
+=================================
+Fine-tuning LLaMA3 with chat data
+=================================
+
+In this tutorial, we'll demystify what prompt templates are and when you'll need them
+and talk about the differences between prompt templating for LLaMA2 and LLaMA3. Then,
+we'll wrap up with a LLaMA3 finetuning example on a custom chat dataset.
+
+.. grid:: 2
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` You will learn:
+
+      * What is a prompt template and how is it different from special tokens
+      * When to use and not to use a prompt template
+      * How to use your own chat dataset to fine-tune LLaMA3
+
+    .. grid-item-card:: :octicon:`list-unordered;1em;` Prerequisites
+
+      * Be familiar with :ref:`configuring datasets<dataset_tutorial_label>`
```

Comment:
I'm not super on the difference between this tutorial and the existing one. Could we split into something like beginner and advanced tutorials? Beginner being low degree of customization, advanced being "here's everything you need to know to set up your own dataset from scratch". Or alternatively make this purely Llama3-focused and go through all those nuances

Reply:
Yeah I feel like there's two main concepts I want to convey in this tutorial: dealing with llama3's new formatting and finetuning on a custom dataset with llama3. I'm thinking of keeping those two in this tutorial for now and see how to dedupe with the dataset tutorial later, do you think some topics should move to the other tutorial?

Reply:
I also think a bit of redundancy is ok or even good, it's helpful to see similar information conveyed through different problems, and this llama3 data tutorial and dataset tutorials approach the concepts different